src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+          ctx.emitEvent(ChartToTSEvent.OpenContextMenu, {
+            event: getParsedEvent(e),
+            clickedPoint: {
+              tuple: dataModel.getPointDetails(dataX, dataY, chartType),
+            },
+          });
+        },
+      },
+
+      plugins: [htmlLegendPlugin],
+    });
+  } catch (e) {
+    logger.error("renderfailed", e);
+    throw e;
+  }
+}
+
+const renderChart = async (ctx: CustomChartContext): Promise<void> => {
+  if (globalChartReference) {
+    globalChartReference.destroy();
+  }
+  try {
+    ctx.emitEvent(ChartToTSEvent.RenderStart);
+    render(ctx);
+  } catch (e) {
+    ctx.emitEvent(ChartToTSEvent.RenderError, {
+      hasError: true,
+      error: e,
+    });
+  } finally {
+    ctx.emitEvent(ChartToTSEvent.RenderComplete);
+  }
+};
+
+(async () => {
+  // populate it for multiple configs, need to populate common x-axis config
+  const ctx = await getChartContext({
+    getDefaultChartConfig: (chartModel: ChartModel): ChartConfig[] => {
+      const cols = chartModel.columns;
+
+      const measureColumns = _.filter(
+        cols,
+        (col) => col.type === ColumnType.MEASURE
+      );
+
+      const attributeColumns = _.filter(
+        cols,
+        (col) => col.type === ColumnType.ATTRIBUTE
+      );
+
+      const axisConfig: ChartConfig[] = [
+        {
+          key: "column",
+          dimensions: [
+            {
+              key: "x",
+              columns: [attributeColumns[0]],
+            },
+            {
+              key: "y",
+              columns: measureColumns.slice(0, 2),
+            },
+          ],
+        },
+        {
+          key: "line",
+          dimensions: [
+            {
+              key: "y",
+              columns: measureColumns.slice(0, 2),
+            },
+          ],
+        },
+        {
+          key: "stacked-column",
+          dimensions: [
+            {
+              key: "y",
+              columns: measureColumns.slice(0, 2),
+            },
+          ],
+        },
+        {
+          key: "scatter",
+          dimensions: [
+            {
+              key: "y",
+              columns: measureColumns.slice(0, 2),
+            },
+          ],
+        },
+      ];
+      return axisConfig;
+    },
+    getQueriesFromChartConfig: (chartConfig: ChartConfig[]): Array<Query> => {
+      const xAxisColumn = chartConfig[0].dimensions.filter(
+        (col) => col.key === "x"
+      )[0].columns;
+      const queries = chartConfig.map(
+        (config: ChartConfig): Query =>
+          _.reduce(
+            config.dimensions,
+            (acc: Query, dimension) => {
+              // we want to avoid adding x axis columns multiple times.
+              if (dimension.key === "x") {
+                return acc;
+              }
+              return {
+                queryColumns: [...acc.queryColumns, ...dimension.columns],
+              };
+            },
+            {
+              queryColumns: [xAxisColumn[0]],
```

Comment:
do we allow multiple x axis columns? if yes, this will break

Reply:
Right now we are not allowing multiple columns in x axis from the configurator itself.
Will add the comment here